samples/drivers/counter/alarm/src/main.c

```diff
@@ -21,7 +21,7 @@ struct counter_alarm_cfg alarm_cfg;
 #define TIMER DT_NODELABEL(tc0)
 #elif defined(CONFIG_COUNTER_MICROCHIP_MCP7940N)
 #define TIMER DT_NODELABEL(extrtc0)
-#elif defined(CONFIG_COUNTER_RTC0)
+#elif defined(CONFIG_COUNTER_NRF_RTC)
```

Comment:
COUNTER_NRF_RTC is a flag that indicates that RTC based counter driver is used because there is at least one instance with that compatible enabled.
So if that implementation is enabled then we are taking `rtc0`.
It would probably be better to use a fixed label in all overlays and take that device.